### DISCUSSION TITLE
Reset Slashes, tools and variables

### DISCUSSION THREAD

**Discussion by jarmex** [olimorris/codecompanion.nvim#511] [General]
Thank you for such a great tool.

It appears that when I open a chat window (and using `#buffer` or `@editor` for example) and I subsequently open a new buffer for editing, the tools referenced within the chat continue to operate on the previously opened buffer rather than the newly opened one. My question is whether there exists a mechanism to reset the tools, variables, or slashes without having to close the chat window entirely.

The reason I inquire about this feature stems from the fact that the chat history can serve as a valuable reference for me while working on another buffer. However, I would like the tools to operate on the new buffer, rather than the previously opened one. I am hoping for clarification or guidance on how to achieve this. Thanks

**Comment by olimorris** [comment]
Hi @jarmex. Very timely points you've raised. I'll shortly be launching the ability to modify references like buffers or slash commands (you'll be able to delete them from the buffer) and tools will no doubt follow shortly after that. So you will be able to modify what everything points to. Appreciate right now it's really messy and basically requires `gx` or a new chat buffer.
